express-post-json/server.ts

```diff
@@ -0,0 +1,32 @@
+import express from 'express';
+
+type Grade = { id: number; name: string; course: string; score: number };
+type Grades = Record<number, Grade>;
+
+let id = 1;
+
+const grades: Grades = {};
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const gradeArr = [];
+  for (const key in grades) {
+    gradeArr.push(grades[key]);
+  }
+  res.json(gradeArr);
+});
+
+app.use(express.json());
```

Comment:
Typically we do this at the top of the file, right after creating app. This is because it's easy to get put endpoints that require it before the call, and that leads to hard to find bugs. Also, the JSON middleware does nothing if the content-type is not 'application/json' so there's no harm in having it run on every request.